Frontend/app/src/main/java/io/happybugs/happybugs/ReportActivity.java

```diff
@@ -0,0 +1,143 @@
+                        editTexts.whenText, editTexts.whoText, editTexts.detailsText);
+                enableBoxes(editTexts.whatText, checkBoxes.whatCheck, buttons.saveBtn);
+                enableView(editTexts.whatText, views.whatView,
+                        views.whereView, views.whenView, views.whoView, views.detailsView);
+                break;
+            case R.id.whereBtn:
+                enableTextView(editTexts.whereText, editTexts.whatText,
+                        editTexts.whenText, editTexts.whoText, editTexts.detailsText);
+                enableBoxes(editTexts.whereText, checkBoxes.whereCheck, buttons.saveBtn);
+                enableView(editTexts.whereText, views.whereView,
+                        views.whatView, views.whenView, views.whoView, views.detailsView);
+                break;
+            case R.id.whenBtn:
+                enableTextView(editTexts.whenText, editTexts.whatText,
+                        editTexts.whereText, editTexts.whoText, editTexts.detailsText);
+                enableBoxes(editTexts.whenText, checkBoxes.whenCheck, buttons.saveBtn);
+                enableView(editTexts.whenText, views.whenView,
+                        views.whatView, views.whereView, views.whoView, views.detailsView);
+                break;
+            case R.id.whoBtn:
+                enableTextView(editTexts.whoText, editTexts.whatText,
+                        editTexts.whereText, editTexts.whenText, editTexts.detailsText);
+                enableBoxes(editTexts.whoText, checkBoxes.whoCheck,buttons.saveBtn);
+                enableView(editTexts.whoText, views.whoView,
+                        views.whatView, views.whereView, views.whenView, views.detailsView);
+                break;
+            case R.id.detailsBtn:
+                enableTextView(editTexts.detailsText, editTexts.whatText,
+                        editTexts.whereText, editTexts.whenText, editTexts.whoText);
+                enableBoxes(editTexts.detailsText, checkBoxes.detailsCheck, buttons.saveBtn);
+                enableView(editTexts.detailsText, views.detailsView,
+                        views.whatView, views.whereView, views.whenView, views.whoView);
+                break;
+        }
+    }
+
+    //Open and collapse edit text on button click
+    protected void enableTextView(TextView tv1, TextView tv2,
+                                  TextView tv3, TextView tv4, TextView tv5){
+        if (tv1.getVisibility() == View.GONE) {
+            tv2.setVisibility(View.GONE);
+            tv3.setVisibility(View.GONE);
+            tv4.setVisibility(View.GONE);
+            tv5.setVisibility(View.GONE);
+            tv1.setVisibility(View.VISIBLE);
+        } else {
+            tv1.setVisibility(View.GONE);
+        }
+    }
+
+    //Enable checkbox and save button when questions answered
+    protected void enableBoxes(final EditText et, final CheckBox cb, final Button saveBtn){
```

Comment:
et or cb is too short and readers can't figure out what it means until they come back up here. editText or checkBox seems long, but I think it's more helpful in this case.